packages/core/src/components/carousel/carousel.tsx

```diff
@@ -1,142 +1,201 @@
-  @Prop({ mutable: true })
-  pagination?: boolean = true
-  @Prop({ mutable: true })
-  paginationClickable?: boolean
-  @Prop({ mutable: true })
-  paginationType?: PaginationOptions['type'] | 'thumbnails' = 'bullets'
-  @Prop({ mutable: true })
-  slidesPerGroup?: number | string = 1
-  @Prop({ mutable: true })
-  slidesPerView?: number | string = 1
-  @Prop({ mutable: true })
-  spaceBetween?: number = 0
-  @Prop({ mutable: true })
-  speed?: number
-  @Prop({ mutable: true })
-  thumbnailImages?: string = ''
-  @Prop({ mutable: true })
-  videoIcons?: boolean = false
-  @Prop({ mutable: true })
-  navigationButtonSize?: 'medium' | 'xxlarge' = 'medium'
-  @Prop({ mutable: true })
-  watchSlidesProgress?: boolean = false
+  @Prop() loop: boolean = false
+  @Prop() autoplay: number
+  @Prop() thumbnails: string[]
+  @Prop() hasNavigation = true
+  @Prop() hasPagination = true
+
+  @State() currentIdx: number = 0
+
+  @Element() element: HTMLElement
+
+  touchStartX = 0
+  touchEndX = 0
+
+  get currentIndex() {
+    return this.currentIdx
+  }
+
+  set currentIndex(value: number) {
+    if (!this.carouselWrapper) return
+
+    this.currentIdx = value
+    this.carouselWrapper.style.transform = `translateX(-${this.currentIndex * 100}%)`
+  }
+
+  carouselWrapper: HTMLElement
+  carouselItems: NodeListOf<HTMLElement>
+  nextButton: HTMLButtonElement
+  prevButton: HTMLButtonElement
+
+  componentWillLoad() {
+    this.carouselItems = this.element.querySelectorAll('atom-carousel-item')
+
+    if (this.carouselItems?.length <= 1) {
+      this.loop = false
+      this.hasNavigation = false
+      this.hasPagination = false
+    }
+  }
 
   componentDidLoad() {
-    this.swiperEl = this.host.querySelector('swiper-container')
-
-    const children = this.host.querySelectorAll('atom-carousel-item')
-
-    children.forEach((child) => {
-      const slide = child.querySelector('swiper-slide')
-
-      if (slide) {
-        this.swiperEl.appendChild(slide)
-        this.host.removeChild(child)
-      }
-    })
-
-    this.swiperEl.swiper?.on('slideChange', () => {
-      this.atomChange.emit()
-    })
-    this.swiperEl.swiper?.on('navigationPrev', () => {
-      this.atomClickPrev.emit()
-    })
-    this.swiperEl.swiper?.on('navigationNext', () => {
-      this.atomClickNext.emit()
-    })
-
-    let params: { pagination: PaginationOptions } = this.pagination && {
-      pagination: {
-        type:
-          this.paginationType === 'thumbnails' ? 'custom' : this.paginationType,
-        clickable: this.paginationClickable,
-      },
+    this.carouselWrapper =
+      this.element.shadowRoot.querySelector('.carousel-wrapper')
+    this.nextButton = this.element.shadowRoot.querySelector('.navigation--next')
+    this.prevButton = this.element.shadowRoot.querySelector('.navigation--prev')
+
+    this.carouselWrapper.addEventListener('touchstart', this.handleTouchStart)
+    this.carouselWrapper.addEventListener('touchend', this.handleTouchEnd)
```

Comment:
Doubt: What is the difference between `touchstart` and `click`?

Reply:
`touchstart` works just for touch screen devices, in this case, it is possible change the slide position simulating a swiper because we can check the position of the start and end of the touch

```js
  handleTouchStart = (event: TouchEvent) => {
    this.touchStartX = event.touches[0].clientX
  }

  handleTouchEnd = (event: TouchEvent) => {
    this.touchEndX = event.changedTouches[0].clientX
    this.handleSwipe()
  }

  handleSwipe = () => {
    const lastIndex = this.carouselItems.length - 1
    const isSwipeLeft = this.touchEndX < this.touchStartX

    this.currentIndex = isSwipeLeft
      ? Math.min(this.currentIndex + 1, lastIndex)
      : Math.max(this.currentIndex - 1, 0)

    this.showOrHideNavigationButtons()
  }
```